client/packages/programs/src/JsonForms/components/ProgramEvent.tsx

```diff
@@ -92,10 +93,11 @@ const extractAt = (
       case 'after':
         return new Date(encounter?.startDatetime ?? date);
       default:
-        ((_: never) => {})(options.at.encounterStartDatetime as never);
+        ((_: never) => {})(options.at.encounterStartDatetime);
```

Comment:
Am I understanding this correctly? The default case should never occur, but if it does, it returns nothing? 

Why not have it return something at least?

Would this work:
```ts
switch (options.at.encounterStartDatetime) {
      case 'after':
        return new Date(encounter?.startDatetime ?? date);
      case undefined: // fallthrough to the default 'before' value
      case 'before':
      default:
        const before = new Date(
          new Date(encounter?.startDatetime ?? date).getTime() - 1
        );
        return before;
    }
```

Reply:
It's an "exhaustive switch statement". The purpose is to be forced by the compiler to add a new `case` if the input types changes.

For example, currently the type is `'after' | 'before' | undefined'` and we handle all cases, so the default block should never be reached. Typescript already derives correctly that in the default block the type is `never`: `options.at.encounterStartDatetime: never`. To verify/enforce this we have this dummy function with the `never` argument.

If we now change the type to something like: `'after' | 'before' | 'between' | undefined'` the type in the default block is not `never` any more but `between` and we get the wanted compile error because `'between'` can't be passed in as a `never` argument.

Reply:
Yeah, I find it a bit weird to define a function for something that should never happen, but I get the point.

When I last did one of these, I just made the default/never case throw an error, e.g.:
```ts
default:
      // "never" type ensures we will get a *compile-time* error if we are
      // missing a case defined in Endpoints types
      const missingValue: never = endpointKey
      throw new Error('Failed to consider case:' + missingValue)
``` 

Not sure if that's any better?

Reply:
Either way, I think you might still need the `as never`, it still shows as an error in my IDE:

![Screenshot 2023-05-03 at 11 07 50 AM](https://user-images.githubusercontent.com/5456533/235804121-21ede3d4-858d-43a1-b7fc-2fe160cb83c9.png)

Reply:
oh that's weird this didn't show here. `options.at.encounterStartDatetime` shouldn't have type `any`... The never cast would break the check...

> Not sure if that's any better?

Like this one, one line more but IMHO easier to understand, and you get the runtime check as well!